agot-bg-game-server/src/common/ingame-game-state/select-units-game-state/SelectUnitsGameState.ts

```diff
@@ -32,6 +33,15 @@ export default class SelectUnitsGameState<P extends SelectUnitsParentGameState>
         this.possibleUnits = possibleUnits;
         this.count = count;
         this.canBeSkipped = canBeSkipped;
+
+        if (count > possibleUnits.length) {
+            throw new Error("User has to select more units than possible and therefore SelectUnitsGameState will never end!");
```

Comment:
We should add an additional inner `if (canBeSkipped)` and do only a console.warn in case state can be skipped and throw the error if it can't be skipped.

Reply:
I don't mind throwing an error even if the GameState can be skipped because this is a situation that should never arise, and I'd prefer a game to crash if it happens than hanging in a broken state.

It might annoy the users that would run into this, but it would allow us to detect this error sooner.

Reply:
Alright! :)